specifications/EinsteinRiddle/Einstein.tla

```diff
@@ -35,12 +35,14 @@
 
 EXTENDS Naturals, FiniteSets
 
+House == 1..5
+
 \* Note that TLC!Permutations has a Java module override and, thus,
 \* would be evaluated faster.  However, TLC!Permutations equals a
 \* set of records whereas Permutation below equals a set of tuples/
 \* sequences.  Also, Permutation expects Cardinality(S) = 5.
 Permutation(S) == 
```

Comment:
Just by looking at the set of constraints, I think you could define `Permutation` for the general case too:

```tla
Permutation(S) ==
  { p \in [ House -> S ]:
    \A h1 \in House:
      \A h2 \in House:
        h2 /= h1 => p[h2] /= p[h1]
  }
```

You could even replace `House` with `S`, and it should still work in Apalache, as long as `S` is a finite set. Of course, this definition would produce really a lot of constraints, something like $O(|S|^{|House|}*|House|^2)$